src/components/BookInfo.tsx

```diff
@@ -0,0 +1,47 @@
+import { Link, useLocation } from 'react-router-dom'
+import './BookInfo.css'
+const BookInfo = () => {
+  const location = useLocation()
```

Comment:
you should get the book id from the location pathname THEN you should do a GET request from the google books api to retrieve the type safe data

Please create a custom hook to do this operation that can be consumed like this

`const {data, loading, error} = useGetBookFromUrl()`